libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -955,23 +991,34 @@ std::string IRGenerator::deployCode(ContractDefinition const& _contract)
 	if (_contract.isLibrary())
 	{
 		solAssert(ContractType(_contract).immutableVariables().empty(), "");
-		immutables.emplace_back(std::map<std::string, std::string>{
-			{"immutableName"s, IRNames::libraryAddressImmutable()},
-			{"value"s, "address()"}
-		});
-
+		if (!eof)
+			immutables.emplace_back(std::map<std::string, std::string>{
+				{"immutableName"s, IRNames::libraryAddressImmutable()},
+				{"value"s, "address()"}
+			});
+		else
+			t("libraryAddressImmutableOffset", std::to_string(m_context.libraryAddressImmutableOffset()));
 	}
 	else
+	{
 		for (VariableDeclaration const* immutable: ContractType(_contract).immutableVariables())
 		{
 			solUnimplementedAssert(immutable->type()->isValueType());
 			solUnimplementedAssert(immutable->type()->sizeOnStack() == 1);
-			immutables.emplace_back(std::map<std::string, std::string>{
-				{"immutableName"s, std::to_string(immutable->id())},
-				{"value"s, "mload(" + std::to_string(m_context.immutableMemoryOffset(*immutable)) + ")"}
-			});
+			if (!eof)
+				immutables.emplace_back(std::map<std::string, std::string>{
+					{"immutableName"s, std::to_string(immutable->id())},
+					{"value"s, "mload(" + std::to_string(m_context.immutableMemoryOffset(*immutable)) + ")"}
+				});
 		}
+
+		t("auxDataStart", std::to_string(CompilerUtils::generalPurposeMemoryStart));
+		// TODO: Make sure that we cannot use reservedMemory() here
+		t("auxDataSize", std::to_string(m_context.reservedMemorySize()));
```

Comment:
This is in the wrong branch. You set it for the non-library case (with and without EOF). Should be set for the EOF case instead (with and without a library).

Reply:
sure. Forgot to move it